### PR TITLE
[Diffusion] Add Cache-DiT support for SenseNova-U1

### PR DESCRIPTION
**Pull request by AuFlow**
Part of #37742

## Motivation
SenseNova-U1 currently cannot use Cache-DiT acceleration in SGLang.

Unlike conventional diffusion transformers, SenseNova-U1 shares its Qwen3
decoder between prefix processing, Think/text generation, KV-cache updates,
and iterative image denoising. Applying Cache-DiT to the entire decoder would
therefore affect non-denoising paths and could corrupt the prefix or KV-cache
state.
This PR adds request-scoped Cache-DiT support that routes only pure image
denoising forwards through the cached decoder blocks while preserving the
native decoder layers for all other paths.

Key Value & Quantifiable Benefits:
🚀 **2.76× Speedup**: Reduces the mean end-to-end latency for 2048×2048 image generation from 62.26s to 22.54s, with request throughput increasing proportionally by 2.76×.
🎯 **Negligible Accuracy Loss**: Rigorous evaluation on the LongText-Bench dataset shows an absolute score difference of only 0.3 points (from 99.8 to 99.5, a ~0.33% relative difference). This <1% variance is considered virtually lossless in both quantitative metrics and visual quality.
💾 **Minimal Memory Overhead**: Achieves this massive speedup with only a ~144 MB (+0.39%) increase in peak GPU memory, making it highly cost-effective for production environments.

## Summary

Add request-scoped Cache-DiT support for SenseNova-U1 image generation.

- Enable Cache-DiT only for pure image denoising forwards.
- Preserve native Qwen decoder layers for prefix, Think, text, and KV-cache update paths.
- Support both dense Qwen3 and Qwen3-MoE SenseNova backbones.
- Add per-request mount, refresh, unmount, and remount-on-CFG-mode-change lifecycle handling.
- Expose `enable_cache_dit` and `cache_dit_params` through SenseNova-U1 CLI overrides.
- Restrict the initial implementation to DBCache knobs.
- Disable Cache-DiT safely for timestep-gated CFG; full-range CFG is supported.

## Implementation notes

Cache-DiT replaces the decoder `layers` list with a unified wrapper during the denoising forward. The integration:

- Pins the custom adapter to `blocks_name="layers"` to avoid selecting the native-layer alias.
- Handles the wrapper's lack of per-layer `attention_type` by deriving and validating a uniform attention type from the native layers.
- Uses Cache-DiT separate-CFG mode when `guidance_scale > 1`.
- Includes CFG mode in the mounted-cache key so requests switching between CFG and non-CFG remount safely.

## Performance Testing and Accuracy Testing
**Cache-DiT server**
```
export SGLANG_CACHE_DIT_ENABLED=true
sglang serve \
  --model-path sensenova/SenseNova-U1.5-8B-MoT \
  --port 30000
```
**Benchmark command**
```
python -m sglang.multimodal_gen.benchmarks.bench_serving \
  --host 127.0.0.1 \
  --port 30000 \
  --task text-to-image \
  --dataset vbench \
  --num-prompts 10 \
  --max-concurrency 1 \
  --width 2048 \
  --height 2048 \
  --num-inference-steps 50
```
###  Performance  Results

| Metric | Baseline | Cache-DiT | Improvement |
| --- | ---: | ---: | ---: |
| Benchmark duration | 622.57 s | 225.44 s | 2.76× faster |
| Mean latency | 62.26 s | 22.54 s | 63.8% lower |
| Median latency | 62.33 s | 22.71 s | 63.6% lower |
| P90 latency | 62.43 s | 22.91 s | 63.3% lower |
| P95 latency | 62.46 s | 22.94 s | 63.3% lower |
| P99 latency | 62.48 s | 22.96 s | 63.3% lower |
| Request throughput | 0.016 req/s | 0.044 req/s | 2.76× higher |
| Peak memory (max) | 36,508 MB | 36,652 MB | +144 MB (+0.39%) |
| Peak memory (mean) | 36,502.4 MB | 36,638.6 MB | +136.2 MB (+0.37%) |
| Successful requests | 10/10 | 10/10 | — |

Cache-DiT reduces the mean end-to-end latency for 2048 × 2048 image
generation from 62.26 seconds to 22.54 seconds. This represents a **2.76×
speedup** and a **63.8% latency reduction**, with only **144 MB (0.39%)**
additional peak GPU memory.

### Accuracy Testing

###server
```
export SGLANG_CACHE_DIT_ENABLED=true
export SGLANG_CACHE_DIT_FN=4
export SGLANG_CACHE_DIT_BN=2
export SGLANG_CACHE_DIT_WARMUP=8
export SGLANG_CACHE_DIT_RDT=0.10
export SGLANG_CACHE_DIT_MC=1

 sglang serve \
  --model-path sensenova/SenseNova-U1.5-8B-MoT \
  --port 30000 \
  --host 0.0.0.0
```

#### Case 1
**Prompt:** `a photo of two pizzas `
**Seed:** `1`

| Baseline | Cache-DiT |
| :---: | :---: |
| 5d059cc4-096a-41f6-8c08-28388bb89c31 | d07e46b9-937e-4a18-b382-7342e875d716 |
#### Case 2
**Prompt:** `一条宁静的乡村道路延伸至地平线，两旁是起伏的绿色山丘和金色田野。道路旁立着一块斑驳的木质交通标志，其油漆略有褪色但依然清晰可见。标志上以粗体大写字母写着“前方野生动物横穿”。在主文本下方，较小的字迹写着“减速慢行，保护自然”。标志固定在一根坚固的立柱上，周围长满高草和随风摇曳的野花。远处，一只鹿谨慎地踏上了道路，增添了与自然的宁静联系。上方的天空蔚蓝如洗，点缀着蓬松的白云，而空气中回荡着鸟鸣的宁静声响。这场景唤起对自然世界的平静与尊重。 `
**Seed:** `1`

| Baseline | Cache-DiT |
| :---: | :---: |
| 65b1bb3c-e0d2-4122-b85f-c1a6ece31ac4 | 155603d1-ed08-4e4a-b31c-0f9b925d68b1 |

#### Accuracy Validation

We evaluated accuracy on the first 10% subset of the LongText-Bench dataset.

| configuration | score | image |
| :---: | :---: |:---: |
| base| 99.8 | image |
| cache dit | 99.5 | image |

Cache DIT shows an absolute score difference of 0.33 points compared with the baseline, corresponding to an approximately 0.33% relative difference. This difference is within 1% and can be considered negligible.

## Tests

- Added Cache-DiT adapter tests for dense and MoE SenseNova Qwen backbones.
- Added generation-stage lifecycle tests for mount, refresh, unmount, and CFG-mode remount.
- Ran:
  - `python python/sglang/multimodal_gen/test/unit/test_cache_dit_integration.py`
  - `python -m compileall ...`
  - `git diff --check`

## Limitations

- Timestep-gated CFG (`cfg_interval != (0, 1)`) currently disables Cache-DiT conservatively.
- Full end-to-end image quality and performance validation requires a Linux/GPU environment with Cache-DiT and a SenseNova checkpoint.

---
### CI States

Latest PR Test (Base): :x: [Run #35500140590](https://github.com/sgl-project/sglang/actions/runs/35500140590)
Latest PR Test (Extra): :x: [Run #35500140348](https://github.com/sgl-project/sglang/actions/runs/35500140348)
Latest PR Test (AMD ROCm 10): :x: [Run #35500140474](https://github.com/sgl-project/sglang/actions/runs/35500140474)

### CODE REVIEW

**Comment by ChefWu551** [comment]
Please also update the usage instructions in the cookbook. You can refer to the [[4.2 Advanced Usage](https://docs.sglang.io/cookbook/diffusion/Qwen-Image/Qwen-Image#4-2-advanced-usage)](https://docs.sglang.io/cookbook/diffusion/Qwen-Image/Qwen-Image#4-2-advanced-usage) section in the Qwen-Image cookbook.

**Comment by ChefWu551** [comment]
Could you provide the current accuracy evaluation results and compare them against the official benchmark results to make sure they are aligned?

**Comment by rwang5203** [comment]
Combined with #39081, edits with `guidance_scale < 1` share cached residuals across conditioning branches. Consider deriving cache mode from the editing branch schedule.

**Comment by AuFlow** [comment]
Addressed by deriving the Cache-DiT mode from SenseNova's exact guidance branch schedule instead of guidance_scale > 1.
For image edits with guidance_scale < 1 and the default img_cfg_scale = 1, the derived profile is condition + image_condition, so Cache-DiT is mounted with has_separate_cfg=True and the two branches use isolated residual state. Three-branch and timestep-gated schedules fail closed.
I also added branch-matrix and residual-isolation regression tests and verified the combined path with #39081.

File: python/sglang/multimodal_gen/runtime/models/sensenova_u1/neo_unify/modeling_neo_chat.py [hunk 1]
```diff
@@ -6,6 +6,11 @@
 
 import torch.utils.checkpoint
 import transformers
+from sglang.multimodal_gen.configs.sensenova_u1 import (
+    DEFAULT_IMG_CFG_SCALE,
+    SenseNovaGuidanceProfile,
+    derive_guidance_profile,
+)
```
Comment (rwang5203) [inline python/sglang/multimodal_gen/runtime/models/sensenova_u1/neo_unify/modeling_neo_chat.py]: The new import blocks fail the repository's isort hook. Consider applying the configured sorter to all three modified import sections.

Comment (AuFlow) [reply python/sglang/multimodal_gen/runtime/models/sensenova_u1/neo_unify/modeling_neo_chat.py]: Thanks for pointing that out — I've already fixed it.
